test/functional/tests/lazy_writes/test_flush_huge_dirty_data.py

```diff
@@ -0,0 +1,169 @@
+
+    with TestRun.step(f"Add core with {fs.name} filesystem to cache and mount it."):
+        core_dev.create_filesystem(fs)
+        core = cache.add_core(core_dev)
+        core.mount(mnt_point)
+
+    with TestRun.step("Disable cleaning and sequential cutoff."):
+        cache.set_cleaning_policy(CleaningPolicy.nop)
+        cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+
+    with TestRun.step("Create test file"):
+        test_file_main = File.create_file("/tmp/test_file_main")
+        fio = (
+            Fio().create_command()
+            .io_engine(IoEngine.libaio)
+            .read_write(ReadWrite.write)
+            .block_size(bs)
+            .direct()
+            .io_depth(256)
+            .target(test_file_main.full_path)
+            .size(file_size)
+        )
+        fio.default_run_time = timedelta(hours=4)  # timeout for non-time-based fio
+        fio.run()
+        test_file_main.refresh_item()
+
+    with TestRun.step("Validate test file and read its md5 sum."):
+        if test_file_main.size != file_size:
+            TestRun.fail("Created test file hasn't reached its target size.")
+        test_file_md5sum_main = test_file_main.md5sum()
+
+    with TestRun.step("Write data to exported object."):
+        test_file_copy = test_file_main.copy(mnt_point + "test_file_copy")
+        test_file_copy.refresh_item()
+        sync()
+
+    with TestRun.step("Check if dirty data exceeded 400 GiB."):
+        minimum_4KiB_blocks = int(Size(400, Unit.GiB).get_value(Unit.Blocks4096))
+        if int(cache.get_statistics().usage_stats.dirty) < minimum_4KiB_blocks:
+            TestRun.fail("There are not enough dirty data in cache!")
+
+    with TestRun.step("Unmount core and stop cache with flush."):
+        core.unmount()
+        try:
+            # this operation could take few hours, depending on core disk
+            output = TestRun.executor.run(stop_cmd(str(cache.cache_id)), timedelta(hours=12))
+            if output.exit_code != 0:
+                raise CmdException("Failed to stop cache.", output)
+        except CmdException:
+            TestRun.fail("Stopping cache with flush failed!")
+
+    with TestRun.step("Mount core device and check md5 sum of test file copy."):
+        core_dev.mount(mnt_point)
+        if test_file_md5sum_main != test_file_copy.md5sum():
+            TestRun.LOGGER.error("Md5 sums should be equal.")
+
+    with TestRun.step("Delete test files."):
+        test_file_main.remove(True)
+        test_file_copy.remove(True)
+
+    with TestRun.step("Unmount core device."):
+        core_dev.unmount()
+        remove(mnt_point, True, True, True)
+
+
+@pytest.mark.parametrize("cache_mode", CacheMode.with_traits(CacheModeTrait.LazyWrites))
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_flush_over_400_gibibytes_without_fs(cache_mode):
```

Comment:
maybe rename it to raw_device because it will be easier to find those tests which are running on raw device. Please update test title as well